definitions/features/hammer.rb

```diff
@@ -99,24 +100,24 @@ def ask_password_and_check(custom_config)
 
   def config_error
     raise ForemanMaintain::HammerConfigurationError, 'Hammer configuration failed: '\
-                  "Is the admin password correct? (it was stored in #{custom_config_file})" \
+                  'Is the admin credential from the file' \
+                  " #{custom_config_file} correct?\n" \
                   'Is the server down?'
   end
 
   def on_missing_password(custom_config)
     if admin_password_missing?
       msg = 'Admin password was not found in hammer configs. Looking into installer answers'
       logger.info(msg)
-      custom_config[:foreman][:password] = password_from_answers
+      custom_config[:foreman][:password] = password_from_answers(custom_config[:foreman][:username])
     end
     save_config_and_check(custom_config)
     custom_config
   end
 
   def admin_password_missing?
     configuration[:foreman][:password].nil? ||
-      configuration[:foreman][:password].empty? ||
-      configuration[:foreman][:username] != 'admin'
+      configuration[:foreman][:password].empty?
```

Comment:
I think, here condition for username comparison is also required. 
Instead of comparing `configuration[:foreman][:username] != admin`, could you please simply modify the condition as `configuration[:foreman][:username] != username` ?

Reply:
Thanks @kgaikwad for helping me in fixing tests.
I have updated the PR and all the tests are fixed now.